test/unwind.js

```diff
@@ -0,0 +1,74 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+
+describe('unwind', function() {
+  var object = { name: 'alice',
+    hobbies: ['Golf', 'Hacking'],
+    colors: ['red', 'green']
+  };
+  var quiz = {
+    title: 'What is the best programming language?',
+    options: {0: 'Python', 1: 'Java', 2: 'JavaScript', 3: 'C++'},
+    ans: 'Correct Answer is C'
+  };
+
+
+  it('returns list of destructed object if key is present as list in the object', function() {
+    eq(R.unwind('hobbies', object), [
+      {name: 'alice', hobbies: 'Golf', colors: ['red', 'green']},
+      {name: 'alice', hobbies: 'Hacking', colors: ['red', 'green']}
+    ]);
+    eq(R.unwind('colors', object), [
+      {name: 'alice', hobbies: ['Golf', 'Hacking'], colors: 'red'},
+      {name: 'alice', hobbies: ['Golf', 'Hacking'], colors: 'green'}
+    ]);
+  });
+
+  it('returns list of original object if key is not present in the object', function() {
+    eq(R.unwind('hobby', object), [object]);
+  });
+
+  it('returns list of original object if key is present in the object but does not have iterable value to be destructed', function() {
+    eq(R.unwind('passtimes', object), [object]);
+    eq(R.unwind('options', quiz), [quiz]);
+    eq(R.unwind('hobbies', { name: 'Berney', hobbies: NaN}), [{ name: 'Berney', hobbies: NaN}]);
+  });
+
+  it('key can be any value that can form key of the object', function() {
```

Comment:
Here, ```[ -ve number ]```  is evaluated as string. Our idea was that -ve number cannot form the key of object. 
```js
var object = {
      [-1]: ['Python', 'Java', 'JavaScript', 'C++']
    }
unwind(-1)(object) //=>
// [{"-1": ["Python", "Java", "JavaScript", "C++"]}]
````
For this to work when key should be string
```js
unwind('-1')(object) //=>
// [{"-1": "Python"}, {"-1": "Java"}, {"-1": "JavaScript"}, {"-1": "C++"}]
```

Reply:
I'm fairly certain that this is the same behavior with a negative or positive number.

In [my test][rr], 

```
unwind (-1) (object) //=> [{"-1": "Python"}, {"-1": "Java"}, {"-1": "JavaScript"}, {"-1": "C++"}]
```

Which is very similar to how it would work with a positive number:

```js
const object = {
  [1]: ['Python', 'Java', 'JavaScript', 'C++']
}

unwind (1) (object) //=> [{"1": "Python"}, {"1": "Java"}, {"1": "JavaScript"}, {"1": "C++"}]
```

Objects do not have numeric keys; all keys are strings or Symbols.  But when you try to get or set with a numeric key, it is automatically converted to a string for you.

The main point, though, is that this test is unnecessary.  It's not really testing `unwind`, but basic capabilities of Javascript objects.

  [rr]: https://tinyurl.com/y9slqfj2